packages/react-components/react-virtualizer/src/hooks/useVirtualizerMeasure.ts

```diff
@@ -97,3 +97,140 @@ export const useStaticVirtualizerMeasure = (
     scrollRef,
   };
 };
+
+/**
+ * React hook that measures virtualized space dynamically to ensure optimized virtualization length.
+ */
+export const useDynamicVirtualizerMeasure = (
```

Comment:
Let's improve the typing for both of these hooks and make the element type generic

```ts
useDynamicVirtualizerMeasure = <TElement extends HTMLElement>(): { scrollRef: (instance: TElement | null) }
```